src/main/java/chatbot/task/Task.java

```diff
@@ -0,0 +1,116 @@
+package chatbot.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class which consist of the detail needed for a Task.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * constructor for this class.
+     *
+     * @param description String for description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * get the status of the task.
+     *
+     * @return String to should the status of the task
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Get description of the task.
+     *
+     * @return description
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * mark task as done
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * unmark the task
+     */
+    public void unMark() {
+        isDone = false;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    /**
+     * convert task to string to save to file
+     *
+     * @return task in String format
+     */
+    public String toFileFormat() {
+        String done = isDone ? "1" : "0";
+        return String.format("T | %s | %s", done, description);
+    }
+
+    /**
+     * Read the task saved in file which is in String format and convert it back as task form
+     *
+     * @param line the line of task
+     * @return task in Task format
+     */
+    public static Task fromFileFormat(String line) {
+        String[] parts = line.split("\\|");
+
+        // Error handling for incorrect line format
+        if (parts.length < 2) {
+            // Handle this case appropriately
+            return null;
+        }
+
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1");
+        String description = parts[2].trim();
+
+        Task task = null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+        switch (type) {
```

Comment:
Would it be better if the switch and case statements have the same indentations?